### DISCUSSION TITLE
SC126 ROMWBW CP/M COM2

### DISCUSSION THREAD

**Discussion by johnwiggins** [wwarthen/RomWBW#388] [Q&A]
I've been trying to get the second serial port to work under CP/M on an SC126, and it looks like all CP/M devices that I would expect to map to a serial port get redirected to the console port (COM1).

Looking at the CBIOS source, it seems to suggest that IO is mapped to CON: (COM1). Is there a suggested way to patch so I can use com2? (would like to add a serial printer).

**Comment by wwarthen** [comment]
Hi @johnwiggins,

I just checked this on my SC126 and it seems to work fine, so let's figure out what the difference is.

I am using RomWBW v3.4 (latest stable release).  From ZSDOS, I can use `PIP LPT:=CON:` and everything I type on the main console (first serial port) is sent to the second serial port of the SC126.

Which version of RomWBW are you using?
Which operating system are you using?

Thanks,

Wayne

**Comment by johnwiggins** [comment]
Thanks Wayne!

I'm using CP/M 2.2 and RomWBW 3.4.

It turns out that I had a bad MAX232 on the second port. Thank you! Your confirmation that this should work made me dig into the hardware a little more.

Thank you so much!

-john

**Reply by wrljet** [reply]
May I ask, the source of the bad chip?

**Reply by wwarthen** [reply]
Glad to hear it is resolved!

-Wayne

**Comment by johnwiggins** [comment]
It was a MAX232 assembly, purchased on Amazon, one of a multi-pack. Others in the pack work fine, so I don't think it's a problematic seller.

One of the reasons I didn't think to swap it out is because they've been very solid.
